recipe/session/sessionwithjwt/sessionContainer.go

```diff
@@ -21,49 +21,121 @@ import (
+
+			if !ok {
+				return originalSessionClass.UpdateAccessTokenPayloadWithContext(newAccessTokenPayload, userContext)
+			}
+
+			existingJWT := accessTokenPayload[jwtPropertyName.(string)].(string)
+
+			currentTimeInSeconds := uint64(time.Now().UnixNano() / 1000000000) // time in seconds
+
+			claims := jwt.MapClaims{}
+			decodedPayload := map[string]interface{}{}
+			_, _, err := new(jwt.Parser).ParseUnverified(existingJWT, claims)
+			if err != nil {
+				return err
+			}
+			for key, val := range claims {
+				decodedPayload[key] = val
+			}
+
+			jwtExpiry := uint64(decodedPayload["exp"].(float64)) - currentTimeInSeconds
+
+			if jwtExpiry <= 0 {
+				// it can come here if someone calls this function well after
+				// the access token and the jwt payload have expired (which can happen if an API takes a VERY long time). In this case, we still want the jwt payload to update, but the resulting JWT should
+				// not be alive for too long (since it's expired already). So we set it to
+				// 1 second lifetime.
+				jwtExpiry = 1
+			}
+
+			newAccessTokenPayload, err = addJWTToAccessTokenPayload(newAccessTokenPayload, jwtExpiry, originalSessionClass.GetUserIDWithContext(userContext), jwtPropertyName.(string), openidRecipeImplementation, &map[string]interface{}{})
+			if err != nil {
+				return err
+			}
+
+			return originalSessionClass.UpdateAccessTokenPayloadWithContext(newAccessTokenPayload, userContext)
+		},
 		RevokeSession: func() error {
-			return originalSessionClass.RevokeSession()
+			return originalSessionClass.RevokeSessionWithContext(nil)
 		},
 
 		GetSessionData: func() (map[string]interface{}, error) {
-			return originalSessionClass.GetSessionData()
+			return originalSessionClass.GetSessionDataWithContext(nil)
 		},
 
 		UpdateSessionData: func(newSessionData map[string]interface{}) error {
-			return originalSessionClass.UpdateSessionData(newSessionData)
+			return originalSessionClass.UpdateSessionDataWithContext(newSessionData, nil)
 		},
 		GetUserID: func() string {
-			return originalSessionClass.GetUserID()
+			return originalSessionClass.GetUserIDWithContext(nil)
 		},
 		GetAccessTokenPayload: func() map[string]interface{} {
-			return originalSessionClass.GetAccessTokenPayload()
+			return originalSessionClass.GetAccessTokenPayloadWithContext(nil)
 		},
 		GetHandle: func() string {
-			return originalSessionClass.GetHandle()
+			return originalSessionClass.GetHandleWithContext(nil)
 		},
 		GetAccessToken: func() string {
-			return originalSessionClass.GetAccessToken()
+			return originalSessionClass.GetAccessTokenWithContext(nil)
 		},
 		GetTimeCreated: func() (uint64, error) {
-			return originalSessionClass.GetTimeCreated()
+			return originalSessionClass.GetTimeCreatedWithContext(nil)
 		},
 		GetExpiry: func() (uint64, error) {
-			return originalSessionClass.GetExpiry()
+			return originalSessionClass.GetExpiryWithContext(nil)
```

Comment:
No. This should call `originalSessionClass.GetExpiry()` only. Likewise for all other non modified functions..